woocommerce/woocommerce
# Version Bump: Add CLI utility

### All Submissions:






### Changes proposed in this Pull Request:



Contributes to https://github.com/woocommerce/woocommerce/issues/34424

### How to test the changes in this Pull Request:

1. Try bumping to an invalid version `ts-node ./tools/version-bump/index.ts bump woocommerce -v 99999` and see an appropriate error occur.
2. Try bumping to a lower version than current `ts-node ./tools/version-bump/index.ts bump woocommerce -v 6.8.0-beta.1` and see an appropriate error occur.
3. Bump a dev version as if you were updating trunk after a code freeze, `ts-node ./tools/version-bump/index.ts bump woocommerce -v 7.1.0-dev`. Check the diff and note that all updated files will get a regular version `7.1.0` while the main plugin file gets `7.1.0-dev`. The readme top level changelog title also gets updated.
4. Checkout all files to start fresh and bump a minor version `ts-node ./tools/version-bump/index.ts bump woocommerce -v 7.1.1`. Check the diff to make sure changes are appropriate.
5. Checkout all files again and bump to a prerelease version `ts-node ./tools/version-bump/index.ts bump woocommerce -v 7.1.0-beta.2`. 

**Questions**
1. Do the update make sense in each case? 
2. Do the testing instructions correctly assume desired behaviour?
3. So long as the user increases a version, anything goes. Is checking for appropriate bumps worthwhile? I don't think its worth the trouble, especially if we automate this with a Github workflow.

### Other information:




### FOR PR REVIEWER ONLY:
